ImageInLib/ImageInLib/src/noise_generator.c

```diff
@@ -216,57 +216,110 @@ bool saltAndPepper2dNoise_UC(unsigned char * array2DPtr, const size_t xDim, cons
 	return true;
 }
 
-bool saltAndPepper3dNoise_D(dataType ** array3DPtr, const size_t xDim, const size_t yDim,
-	const size_t zDim, float K)
+bool saltAndPepper3dNoise_D(dataType** array3DPtr, const size_t xDim, const size_t yDim,
+	const size_t zDim, dataType density, const dataType pepper)
 {
-	size_t i, k, s;
-	const size_t dim2DK = (int)((xDim * yDim * K) + 0.5);
-	const size_t dim2D = xDim * yDim;
+	size_t i, j, k, l = 0, m, n, xd, s;
+	dataType K = density;
+
+	//checks for correctness of the density (ie: percent noise, on [0,1] of salt & pepper noise
+	if ((K < 0) && (K > 1))
```

Comment:
You can use density directly here, you don not need to do this -> dataType K = density;. Update the PR and will accept it